vllm/model_executor/models/deepseek_v2.py

```diff
@@ -629,14 +629,28 @@ def sparse_attn_indexer(
         assert topk_tokens == 2048, "top_k_per_row assumes size 2048"
         topk_indices = topk_indices_buffer[:num_decode_tokens, :topk_tokens]
 
+        splitWorkThreshold = 200 * 1000
+        aux_indices = torch.empty((0,), dtype=torch.int32, device=logits.device)
+        aux_logits = torch.empty((0,), dtype=torch.float32, device=logits.device)
+        if logits.shape[1] >= splitWorkThreshold:
+            aux_indices = torch.empty(
+                (num_rows, 10 * topk_tokens), dtype=torch.int32, device=logits.device
+            )
+            aux_logits = torch.empty(
+                (num_rows, 10 * topk_tokens), dtype=torch.float32, device=logits.device
+            )
```

Comment:
Maybe this potential allocation could happen within the top_k_per_row_decode op so we can have one source of truth for here and the test?